### ISSUE TITLE
Not linting every file

### ISSUE THREAD

**Issue by adam-lynch** [lesshint/gulp-lesshint#3]
I'm not sure where the actual error is but this lints a few files but not `./path/to/person.less`:

``` javascript
gulp.task('lint-styles', function(done){
    gulp.src('./path/to/*.less')
        .pipe($.lesshint())
        .pipe($.lesshint.reporter())
        .on('end', done);
});
```

It is linted if I do:

``` javascript
gulp.task('lint-styles', function(done){
    gulp.src('./path/to/person.less')
        .pipe($.lesshint())
        .pipe($.lesshint.reporter())
        .on('end', done);
});
```

Or:

``` javascript
gulp.task('lint-styles', function(done){
    gulp.src(['./path/to/avatar.less', './path/to/person.less'])
        .pipe($.lesshint())
        .pipe($.lesshint.reporter())
        .on('end', done);
});
```

**Comment by adam-lynch** [comment]
Ok there's a problem with `.on('end', done)` or `.on('finish', done)` (which we use regularly with other plugins. There must be something different about this plugin. Removing it (and the `done` parameter) fixes the issue. 

This also (eventually) causes my CSS file not to exist in the end. I have something like:

``` javascript
gulp.task('styles', ['link-styles', 'something-else'], function(done){
    //...
        .on('end', done);
});

gulp.task('lint-styles', ['something-else'], function(done){
    gulp.src('./path/to/*.less')
        .pipe($.lesshint())
        .pipe($.lesshint.reporter())
        .on('end', done);
});
```

**Comment by adam-lynch** [comment]
Hmm yeah, definitely... I tried piping into my own thing after the reporter to `throw` an error if the result count was too high but it only seems to hit my function for one file...

``` javascript
var through = require('through');
//...
.pipe($.lesshint.reporter())
.pipe(through(function(file){ 
  console.log(file.path);
  this.emit('data', file);
})
// no 'end'/'finish'
```

Again, this has worked fine with other plugins. Although I'm not as confident about this as I am the more basic examples above.

**Comment by jwilsson** [comment]
@adam-lynch I might have a theory why this is happening, `gulp-lesshint` doesn't push the file downwards in the stream (while for example, `gulp-coffeelint` does). I've pushed a possible fix for it to the `issue-3` branch and I was just wondering if you could check it out before merging it into `master`.

I'm a bit afraid that it might cause some other issue and you seem to have a pretty advanced Gulp setup so if anything breaks we might find it straight away.

**Comment by adam-lynch** [comment]
Will do @jwilsson

**Comment by jwilsson** [comment]
@adam-lynch just wanted to see if you've had the time to test this?

**Comment by jwilsson** [comment]
This is hopefully fixed in `2.0.0-rc` which is installable with `npm install gulp-lesshint@next`.
